client/rest-high-level/src/main/java/org/elasticsearch/client/RestRequestActions.java

```diff
@@ -0,0 +1,524 @@
+                    // or an error for which parsing is completely different. We try to consider the 404 response as a valid one
+                    // first. If parsing of the response breaks, we fall back to parsing it as an error.
+                    throw parseResponseException(e);
+                }
+            }
+            throw parseResponseException(e);
+        }
+
+        try {
+            return responseConverter.apply(response);
+        } catch(Exception e) {
+            throw new IOException("Unable to parse response body for " + response, e);
+        }
+    }
+
+    /**
+     * @deprecated If creating a new HLRC ReST API call, consider creating new actions instead of reusing server actions. The Validation
+     * layer has been added to the ReST client, and requests should extend {@link Validatable} instead of {@link ActionRequest}.
+     */
+    @Deprecated
+    protected <Req extends ActionRequest, Resp> void performRequestAsyncAndParseEntity(Req request,
+                                                                         CheckedFunction<Req, Request, IOException> requestConverter,
+                                                                         RequestOptions options,
+                                                                         CheckedFunction<XContentParser, Resp, IOException> entityParser,
+                                                                         ActionListener<Resp> listener, Set<Integer> ignores) {
+        performRequestAsync(request, requestConverter, options,
+            response -> parseEntity(response.getEntity(), entityParser), listener, ignores);
+    }
+
+    /**
+     * Defines a helper method for asynchronously performing a request.
+     */
+    protected <Req extends Validatable, Resp> void performRequestAsyncAndParseEntity(Req request,
+                                                                           CheckedFunction<Req, Request, IOException> requestConverter,
+                                                                           RequestOptions options,
+                                                                           CheckedFunction<XContentParser, Resp, IOException> entityParser,
+                                                                           ActionListener<Resp> listener, Set<Integer> ignores) {
+        performRequestAsync(request, requestConverter, options,
+            response -> parseEntity(response.getEntity(), entityParser), listener, ignores);
+    }
+
+
+    /**
+     * @deprecated If creating a new HLRC ReST API call, consider creating new actions instead of reusing server actions. The Validation
+     * layer has been added to the ReST client, and requests should extend {@link Validatable} instead of {@link ActionRequest}.
+     */
+    @Deprecated
+    protected <Req extends ActionRequest, Resp> void performRequestAsync(Req request,
```

Comment:
I think these should be public now?

Reply:
I will public'ize them once i move the package, subsequent PR to come.